### PR TITLE
Simple merge and SBN merge changes

### PR DESCRIPTION
**Pull request by bear-is-asleep**
Makes simple merge modules which combines `MCParticle` vectors, with the option to set mother IDs. Also adds a simple check to `MergeSimSourcesSBN` which checks that `MCParticle` is also merged if the associations are merged.

Replaces [this PR](https://github.com/SBNSoftware/sbncode/pull/404) so we can combine simple and sbn merge into a single PR.

### CODE REVIEW

File: sbncode/LArG4/SimpleMerge_module.cc [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  }
+  fResetMotherID = fResetMotherIDs.size() > 0;
+}
+
+void sbn::SimpleMerge::produce(art::Event& e)
+{
+  auto partCol = std::make_unique<std::vector<simb::MCParticle>>();
+  for (auto const& [i_source,input_label] : util::enumerate(fInputSourcesLabels)) {
+    auto parts = e.getValidHandle<std::vector<simb::MCParticle>>(input_label);
+    for (auto part : *parts) {
```
Comment (PetrilloAtWork) [inline sbncode/LArG4/SimpleMerge_module.cc:78]: We can skip the handle:
```suggestion
    auto const& parts = e.getProduct<std::vector<simb::MCParticle>>(input_label);
    for (auto part : parts) {
```

File: sbncode/LArG4/SimpleMerge_module.cc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      if (fResetMotherID){
+        for (auto const& offset : fResetMotherIDs) {
```
Comment (PetrilloAtWork) [inline sbncode/LArG4/SimpleMerge_module.cc:80]: Given that `offset` is a fundamental type (`int`?) we can avoid the reference indirection:
```suggestion
        for (int offset : fResetMotherIDs) {
```

**Review by PetrilloAtWork — APPROVED** [review]
I left a couple of C++/_art_ style comments for @bear-is-asleep , but the PR can definitely go in.

**Comment by bear-is-asleep** [comment]
> I left a couple of C++/_art_ style comments for @bear-is-asleep , but the PR can definitely go in.

Thanks for your suggestions! I've updated the PR with your suggestions and it builds and runs successfully.

**Comment by bear-is-asleep** [comment]
trigger build

**Review by PetrilloAtWork — APPROVED** [review]
Plain approved, for once.

**Pull request merged**